dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -0,0 +1,108 @@
+﻿#include "pch.h"
+#include "ToastNotificationManager.h"
+#include "Microsoft.Windows.ToastNotifications.ToastNotificationManager.g.cpp"
+#include <winrt/Windows.ApplicationModel.Core.h>
+#include <winrt/Windows.Storage.h>
+#include <winrt/base.h>
+#include "ToastActivationCallback.h"
+#include "externs.h"
+#include "ToastNotificationUtility.h"
+#include <frameworkudk/pushnotifications.h>
+
+static winrt::event<winrt::Windows::Foundation::EventHandler<winrt::Microsoft::Windows::ToastNotifications::ToastActivatedEventArgs>> g_toastHandlers;
+
+winrt::event<winrt::Windows::Foundation::EventHandler<winrt::Microsoft::Windows::ToastNotifications::ToastActivatedEventArgs>>& GetToastHandlers()
+{
+    return g_toastHandlers;
+}
+
+namespace winrt
+{
+    using namespace winrt::Windows::UI;
+    using namespace winrt::Windows::Foundation;
+    using namespace Windows::ApplicationModel::Core;
+}
+
+namespace winrt::Microsoft::Windows::ToastNotifications::implementation
+{
+    static wil::unique_com_class_object_cookie s_toastcomActivatorRegistration;
+
+    winrt::Microsoft::Windows::ToastNotifications::ToastNotificationManager ToastNotificationManager::Default()
+    {
+        static auto toastNotificationManager{ winrt::make<ToastNotificationManager>() };
+        return toastNotificationManager;
+    }
+
+    void ToastNotificationManager::RegisterActivator(winrt::Microsoft::Windows::ToastNotifications::ToastActivationInfo const&  details)
+    {
+        THROW_HR_IF_MSG(E_INVALIDARG, s_toastcomActivatorRegistration, "ComActivator already registered.");
+
+        THROW_HR_IF_NULL(E_INVALIDARG, details);
+
+        std::wstring storedComActivatorString;
+        if (!AppModel::Identity::IsPackagedProcess())
+        {
+            storedComActivatorString = RegisterComActivatorGuidAndAssets(details);
```

Comment:
As a general pattern, it's weird to call Register in an Unregister function. What you really need is a helper called GetActivatorGuid that used by this method and RegisterComActivatorGuidAndAssets.

Reply:
The comment is meant for the UnregisterActivator API below